contributors/design-proposals/hugepages.md

```diff
@@ -0,0 +1,282 @@
+node will treat pre-allocated huge pages similar to other system reservations
+and reduce the amount of `memory` it reports using the following formula:
+
+```
+[Allocatable] = [Node Capacity] - 
+ [Kube-Reserved] - 
+ [System-Reserved] - 
+ [Pre-Allocated-HugePages * HugePageSize] -
+ [Hard-Eviction-Threshold]
+```
+
+The following represents a machine with 10Gi of memory.  1Gi of memory has been
+reserved as 512 pre-allocated huge pages sized 2MB.  As you can see, the
+allocatable memory has been reduced to account for the amount of huge pages
+reserved.
+
+```
+apiVersion: v1
+kind: Node
+metadata:
+  name: node1
+...
+status:
+  capacity:
+    memory: 10Gi
+    hugepages.2MB: 512
+  allocatable:
+    memory: 9Gi
+    hugepages.2MB: 512
+...  
+```
+
+## Pod Specification
+
+A pod must make a request to consume pre-allocated huge pages using the resource
+`hugepages.<hugepagesize>` whose quantity is a positive number of pages. The
+request and limit for `hugepages.<hugepagesize>` must match until such time the
+node provides more reliability in handling overcommit (potentially via
+eviction).  An application that consumes `hugepages.<hugepagesize>` is not in
+the `BestEffort` QoS class.
+
+If a pod consumes huge pages via `shmget`, it must run with a supplemental group
+that matches `/proc/sys/vm/hugetlb_shm_group` on the node.  Configuration of
+this group is outside the scope of this specification.
+
+In order to consume huge pages backed by the `hugetlbfs` filesystem inside the
+specified container in the pod, it is helpful to understand the set of mount
+options used with `hugetlbfs`.  For more details, see "Using Huge Pages" here:
+https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt
+
+```
+mount -t hugetlbfs \
+	-o uid=<value>,gid=<value>,mode=<value>,pagesize=<value>,size=<value>,\
+	min_size=<value>,nr_inodes=<value> none /mnt/huge
+```
+
+The proposal recommends a new volume type `hugePages`.  It will support a single
+`pageSize` option that must be provided by users.  The `pageSize` option aligns
+with the `pagesize` argument in the above mount.  It is used to specify the huge
+page size and pool that is used for accounting.  To write into this volume, the
+pod must make a request for huge pages that match this `pagesize`, otherwise its
+usage will be limited to 0 bytes.  No other option from the mount is carried
+over to the `hugePages` volume type.  The `size` option is defaulted to sum of
+`hugepages.<hugepagesize>` requests made by each container for the specified
+`pagesize`.  This keeps the behavior consistent with memory backed `emptyDir`
+volumes whose usage is ultimately constrained by the pod cgroup sandbox memory
+settings.  The `min_size` option is omitted as its not necessary. It's possible
+we could have extended `emptyDir` to support an additional option for
+`medium=Memory` but extend it to support a `pageSize` parameter, but this
+proposal prefers to create an additional volume type instead.
```

Comment:
Care to explain why?  I am open to discussion, but I want to know the rationale here.

@saad-ali